queries.sql

```diff
@@ -7,4 +7,89 @@ SELECT name, date_of_birth from animals WHERE name IN  ('Agumon','Pikachu');
 SELECT name, escape_attempts from animals WHERE weight_kg BETWEEN '10.4' AND '17.3';
 SELECT * from animals WHERE neutered = true;
 SELECT * from animals WHERE name !='Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- transactions
+-- update species column to unspecified and rollback.
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK;
+
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+-- Commit the transaction.
+COMMIT;
+
+-- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+-- Delete all animals born after Jan 1st, 2022.
+BEGIN;
+
+DELETE FROM animals
+WHERE date_of_birth > DATE '2022-01-01';
+
+-- Create a savepoint for the transaction.
+SAVEPOINT before_weight_update;
+
+-- Update all animals' weight to be their weight multiplied by -1.
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+-- Rollback to the savepoint
+ROLLBACK TO before_weight_update;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+COMMIT;
+
+-- How many animals are there?
+SELECT COUNT(id)
+FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(id)
+FROM animals
+WHERE escape_attempts = 0;
```

Comment:
- [ ] I could not see this query, and the one I saw in the place does not give the correct result to it. I will encourage you to rerun it.

![image](https://github.com/Bifabig/animals-table/assets/86200356/3eb7c9f4-7a2b-4e9f-b54c-264df1811917)

Reply:
Hi @Olamarx,

Thanks for the comments. Screenshots have been corrected!